### PR TITLE
Laboratory work #1, Andrey Klimov - 23FPL1

### PR DESCRIPTION
**Pull request by QuietHellsPage**

### CODE REVIEW

**Comment by QuietHellsPage** [comment]
Hello! As far as I'm concerned, we have to use English here. I have several questions about my code. First of all, I have added  "from collections import Counter" to my code(now I don't use it and will delete it), but there are some others libraries like "from idlelib.iomenu import encoding", "from os.path import split" etc and I don't remember if they were in main.py and in start.py before changes. Do I have to delete these libraries because I don't use them and barely know some of them, I just don't know how they appeared...Secondly, before realizing that we are ought to write our code in main.py I made some changes in start.py, now I understand that I didn't have to make them but as in last question I don't remember if my code in start.py is right now, I think there are some errors. Is it possible to upload the first, right version of start.py using some features of GitHub? And my last question is about my function "compare profiles". Is it need to create some cycle to check if there is the key 'name' in both profiles or it's enough to check types of profiles? I think there is no point in checking if there are elements in 'freq' because such checking cycle was in some function before 'compare profiles', but there are no checks for key 'name' in other functions. Is it needed to create a function with checking it? Thank you for your answer!

**Comment by QuietHellsPage** [comment]
Sorry, I have already found answers for all my questions.

File: lab_1_classify_by_unigrams/main.py [hunk 1]
```diff
@@ -34,6 +41,14 @@ def calculate_frequencies(tokens: list[str] | None) -> dict[str, float] | None:
 
     In case of corrupt input arguments, None is returned
     """
+    if (not isinstance(tokens, list) or
+            not all(isinstance(stroka, str) for stroka in tokens) or not tokens):
+        return None
+    quantity = len(tokens)
```
Comment (Vasilisa-Blyudova) [inline lab_1_classify_by_unigrams/main.py:47]: don't create a separate variable if you only use it once

**Review by Vasilisa-Blyudova — COMMENTED** [review]
Great job! Increase the rating to 10!

File: lab_1_classify_by_unigrams/main.py (hunk 1 above)
Comment (QuietHellsPage) [reply lab_1_classify_by_unigrams/main.py]: Fixed

**Comment by QuietHellsPage** [comment]
Sorry, I have 2 errors in start.py and don't know how to fix them: I001 isort found an import in the wrong position and the import style check also wants me to add some function, but I use only 2 functions...

**Comment by QuietHellsPage** [comment]
I have corrected all the errors! Waiting for your approval!

File: lab_1_classify_by_unigrams/main.py [hunk 2]
```diff
@@ -7,6 +7,7 @@
 
 
 def tokenize(text: str) -> list[str] | None:
+
```
Comment (Vasilisa-Blyudova) [inline lab_1_classify_by_unigrams/main.py:10]: rm whitespace

**Review by Vasilisa-Blyudova — COMMENTED** [review]
Only 8? I'm sure you can get a 10!

File: lab_1_classify_by_unigrams/main.py (hunk 2 above)
Comment (QuietHellsPage) [reply lab_1_classify_by_unigrams/main.py]: fixed

**Pull request closed without merging**